src/storage/ring_buffer.rs

```diff
@@ -0,0 +1,134 @@
+use managed::Managed;
+
+/// A trait for setting a value to the default state.
+///
+/// In-palace analog of Default.
+/// Used by RingBuffer for initializing a storage.
+pub trait Resetable {
```

Comment:
`Resettable`, and let's move it to `smoltcp::storage` since it is not really specific to ring buffers.